packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3480,11 +3480,11 @@ class SemanticsNode with DiagnosticableTreeMixin {
       for (int i = 0; i < childCount; i += 1) {
         childrenInTraversalOrder[i] = sortedChildren[i].id;
       }
-      // _children is sorted in paint order, so we invert it to get the hit test
+      // _children is sorted in inversed paint order, so we use it to get the hit test
       // order.
       childrenInHitTestOrder = Int32List(childCount);
-      for (int i = childCount - 1; i >= 0; i -= 1) {
-        childrenInHitTestOrder[i] = _children![childCount - i - 1].id;
+      for (int i = 0; i < childCount; i += 1) {
+        childrenInHitTestOrder[i] = _children![i].id;
```

Comment:
This is surprising, does this mean we always get it wrong?

Reply:
after reading the code more I think the original documentation may be wrong

_children is assigned with the `childrenInInversePaintOrder` in `SemanticsNode.updateWith`.

which is called in `_buildSemanticsSubtree` where the children is the same as `_childrenAndElevationAdjustments` order and I think by default the _childrenAndElevationAdjustments order is insertion order which will be RenderObject.visitChildForSemantics, which is the paint order.

so the the original code should be current but just bad name.

Reply:
Just an update. Talked with @chunhtai offline. It seems the hit test order was incorrect all the time. But previously when we haven't applied `OverlayPortal` to any widgets, if we push a new route, the items underneath the new route will be thrown away so the hit test only need to handle one "layer" of items and everything is okay. After `OverlayPortal` is applied to `Autocomplete`,  the items underneath the popup still stay in the tree, and the hit test order starts show problems.